catroid/src/androidTestPhiro/java/org/catrobat/catroid/uiespresso/ui/activity/PhiroMyProjectsActivityTest.java

```diff
@@ -0,0 +1,148 @@
+
+import static org.catrobat.catroid.uiespresso.util.matchers.AdapterViewMatchers.withAdaptedDataAt;
+
+@RunWith(AndroidJUnit4.class)
+public class PhiroMyProjectsActivityTest {
+
+	private static final String TAG = PhiroMyProjectsActivityTest.class.getSimpleName();
+	@Rule
+	public BaseActivityInstrumentationRule<PhiroMainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(PhiroMainMenuActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		removePhiroPrograms();
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void phiroProgramsInitiallyLoadedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		ViewInteraction viewInteraction = onView(withId(android.R.id.list));
+
+		Integer position = 0;
+		for (String program : PhiroMainMenuActivity.phiroProjects) {
+			viewInteraction
+					.check(matches(withAdaptedDataAt(program, ++position)));
+		}
+	}
+
+	@Test
+	public void phiroProgramsRestoredWhenDeletedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		StorageHandler.getInstance().deleteProject(PhiroMainMenuActivity.phiroProjects[0]);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		onView(withId(android.R.id.list))
+				.check(matches(withAdaptedDataAt(PhiroMainMenuActivity.phiroProjects[0], 0)));
+	}
+
+	@Test
+	public void phiroProgramsUnchangedWhenModifiedTest() {
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		addBrickToProject(PhiroMainMenuActivity.phiroProjects[0]);
+
+		onView(withId(android.R.id.home))
+				.perform(click());
+
+		onView(withText(R.string.main_menu_programs))
+				.perform(click());
+
+		checkBrickInProject();
+	}
+
+	private void addBrickToProject(String projectName) {
+		try {
+			ProjectManager.getInstance().loadProject(projectName, InstrumentationRegistry.getTargetContext());
+		} catch (LoadingProjectException e) {
+			Log.e(TAG, "Could not load project: " + e.getMessage());
+		} catch (OutdatedVersionProjectException e) {
+			Log.e(TAG, "project version outdated: " + e.getMessage());
+		} catch (CompatibilityProjectException e) {
+			Log.e(TAG, "project not compatible: " + e.getMessage());
+		}
+
+		Project project = ProjectManager.getInstance().getCurrentProject();
+		project.getDefaultScene().getSpriteList().get(0).getScript(0).addBrick(new NoteBrick("I was added!"));
+	}
+
+	private void checkBrickInProject() {
+		Project project = ProjectManager.getInstance().getCurrentProject();
+		Script script = project.getDefaultScene().getSpriteList().get(0).getScript(0);
+		Brick brick = script.getBrick(script.getBrickList().size() - 1);
+		assertTrue("Brick has been deleted!", brick instanceof NoteBrick);
+	}
+
+	private void removePhiroPrograms() {
+		for (String project : PhiroMainMenuActivity.phiroProjects) {
```

Comment:
In the case of Phiro there may be programs:
There are programs (placed in .zip-Files in the assets-folder) that are extracted when MyProjectsActivity is opened. So this is necessary to ensure that no programs exist in the first place.. It could be done in in teardown, alternatively